u-mode/src/main.rs

```diff
@@ -26,6 +26,29 @@ struct UmodeTask {
 }
 
 impl UmodeTask {
+    // (Test) Print String from U-mode Shared Region
+    //
+    // U-mode Shared Region:
+    //    Contains the data to be printed at the beginning of the area.
+    fn op_print_string(&self, len: usize) -> Result<(), UmodeApiError> {
+        // Print maximum 80 chars.
+        const MAX_LENGTH: usize = 80;
+        let vs_input = self
+            .vslice
+            .get_slice(0, len)
+            .map_err(|_| UmodeApiError::InvalidArgument)?;
+        // Copy input from volatile slice.
+        let mut input = [0u8; MAX_LENGTH];
+        vs_input.copy_to(&mut input[..]);
```

Comment:
Because we're using `VolatileSlice` for the shared buffer, that means we're effectively forced to copy the bytes out in order to operate on the data (unless we go `unsafe`), right? I get wanting to copy the data when passing data from the hypervisor to U-mode, but once we're in U-mode this extra `memcpy()` seems pretty wasteful :/

Reply:
Yes, it is. It's  404 bytes though. So I thought this wouldn't be that much worth of an unsafe, given also the amount of memory the attestation code will move around in U-mode.

Reply:
Yeah that's fine. Was just wondering what sizes to expect here. If we end up getting much larger I'd like to us remove that copy.